src/Components/NavBar/NavItem.tsx

```diff
@@ -54,7 +57,16 @@ export const NavItem: React.FC<NavItemProps> = ({
     >
       <Link href={href} color={hoverColor} underlineBehavior="none">
         <Sans size="3" weight="medium" color={hoverColor}>
-          {children}
+          <Box height={25}>
+            {isFunction(children)
+              ? // NavItem children can be called as renderProps so that contents
+                // can operate on UI behaviors (such as changing the color of an
+                // icon on hover).
+                children({
+                  hover,
+                })
+              : children}
```

Comment:
This is where the renderProps pattern described in [this comment](https://github.com/artsy/reaction/pull/2442/files#r285838759) is implemented.